provider/doc_edits.go

```diff
@@ -15,30 +15,166 @@
 package provider
 
 import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
 	"regexp"
+	"strings"
 
 	"github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge"
 )
 
 func editRules(defaults []tfbridge.DocsEdit) []tfbridge.DocsEdit {
-	return append(defaults, fixupImports)
+	return append(defaults, fixupImports(), applyReplacementsDotJSON())
 }
 
-const tfOrTODO = `([tT]erraform|TODO)`
+func fixupImports() tfbridge.DocsEdit {
+	const tfOrTODO = `([tT]erraform|TODO)`
 
-var inlineImportRegexp = regexp.MustCompile("% " + tfOrTODO + " import")
-var quotedImportRegexp = regexp.MustCompile("`" + tfOrTODO + " import`")
+	var inlineImportRegexp = regexp.MustCompile("% " + tfOrTODO + " import.*")
+	var quotedImportRegexp = regexp.MustCompile("`" + tfOrTODO + " import`")
 
-// (?s) makes the '.' match newlines (in addition to everything else).
-var blockImportRegexp = regexp.MustCompile("(?s)In " + tfOrTODO + " v[0-9]+\\.[0-9]+\\.[0-9]+ and later," +
-	" use an `import` block.*?```.+?```\n")
+	// (?s) makes the '.' match newlines (in addition to everything else).
+	var blockImportRegexp = regexp.MustCompile("(?s)In " + tfOrTODO + " v[0-9]+\\.[0-9]+\\.[0-9]+ and later," +
+		" use an `import` block.*?```.+?```\n")
 
-var fixupImports = tfbridge.DocsEdit{
-	Path: "*",
-	Edit: func(path string, content []byte) ([]byte, error) {
-		content = blockImportRegexp.ReplaceAllLiteral(content, nil)
-		content = inlineImportRegexp.ReplaceAllLiteral(content, []byte("% pulumi import"))
-		content = quotedImportRegexp.ReplaceAllLiteral(content, []byte("`pulumi import`"))
-		return content, nil
-	},
+	return tfbridge.DocsEdit{
+		Path: "*",
+		Edit: func(path string, content []byte) ([]byte, error) {
+			content = blockImportRegexp.ReplaceAllLiteral(content, nil)
+			content = inlineImportRegexp.ReplaceAllFunc(content, func(match []byte) []byte {
+				match = bytes.ReplaceAll(match, []byte("terraform"), []byte("pulumi"))
+				match = bytes.ReplaceAll(match, []byte("Terraform"), []byte("pulumi"))
+				match = bytes.ReplaceAll(match, []byte("TODO"), []byte("pulumi"))
+				return match
+			})
+			content = quotedImportRegexp.ReplaceAllLiteral(content, []byte("`pulumi import`"))
+			return content, nil
+		},
+	}
+}
+
+func applyReplacementsDotJSON() tfbridge.DocsEdit {
+	filePath := "./provider/replacements.json"
+	fileBytes, err := os.ReadFile(filePath)
+	if err != nil {
+		panic(err)
+	}
+	var replacements replacementFile
+	err = json.Unmarshal(fileBytes, &replacements)
+	if err != nil {
+		panic(err)
+	}
+
+	for k, v := range replacements {
+		i := strings.LastIndexByte(k, '/')
+		if i == -1 {
+			continue
+		}
+		delete(replacements, k)
+
+		key := k[i+1:]
+		replacements[key] = append(replacements[key], v...)
+	}
```

Comment:
What's the purpose of this block? We're deleting entries which are actually folders (based on the trailing `/`)?